src/routes/api/issues.ts

```diff
@@ -1,75 +1,67 @@
-        //if the room's section isn't valid / isn't found in database
-        response.status(422).json({ error: "That section doesn't exist, or was inputted incorrectly." });
+    const timeBlock = await TimeBlockModel.findOne({ _id: bookingId });
+    if (!timeBlock) {
+        res.status(400);
         return;
     }
-    //if the section is valid
-    if (typeof message !== 'string') {
-        //checks to ensure that the message is a string type to prevent attacks
-        response.status(422).json({ error: "That message isn't the proper format!" });
+    const section = await SectionModel.findOne({ _id: timeBlock.sectionId });
+    if (!section) {
+        res.status(400);
         return;
     }
-    //if the input type for message is string
-    if (ts === null) {
-        response.status(422).json({ error: "That timestamp isn't the proper format!" });
-        return;
-    }
-    if (user === null) {
-        response.status(422).json({ error: 'User not found!' });
+    const room = await RoomModel.findOne({ _id: section.roomId });
+    if (!room) {
+        res.status(400);
         return;
     }
 
-    const issue = new issueModel({
-        timestamp: ts,
+    const issue = new IssueModel({
+        reportingUserEmail: req.userEmail,
         message: message,
-        status: 'Unresolved',
-        roomId: roomId,
-        sectionId: sectionId,
-        reportingUserId: user.id,
+        status: 'unresolved',
+        bookingId: bookingId,
     });
-    if (room === null) {
-        //if the room isn't valid / isn't found in database
-        response.status(422).json({ error: "That room doesn't exist, or was inputted incorrectly." });
-        return;
-    } else {
-        //if room exists, push the issue id as foreign key for future population in querying
-        room.issues.push(issue.id);
+    issue.save();
+
+    room.issues.push(issue._id);
+    room.save();
+
+    const possibleTimeBlocks = await TimeBlockModel.find({ sectionId: timeBlock.sectionId, startsAt: { $lte: timeBlock.startsAt } });
+    let maxDate = new Date(0);
+    let blamedUsers;
+    for (const possibleTimeBlock of possibleTimeBlocks) {
+        if (possibleTimeBlock.startsAt > maxDate) {
+            maxDate = possibleTimeBlock.startsAt;
+            blamedUsers = possibleTimeBlock.users;
+        }
     }
 
-    //save room and issues
-    let err = false;
-    await room.save().catch((error) => {
-        response.status(400).json({ error });
+    if (!blamedUsers) {
+        res.status(201).json({ issue });
         return;
-    });
-    await issue.save().catch((e) => {
-        console.log(e);
-        response.status(400).json({ error: "The issue couldn't be created!" });
-        err = true;
-    });
+    }
 
-    if (err) {
-        return;
+    for (const blamedUser of blamedUsers) {
+        sendEmail({
+            to: blamedUser,
+            subject: 'Issue Reported',
+            text: `An issue was reported for your booking`,
```

Comment:
Should we include the `message` as well?

Reply:
We discussed this and we decided not to since a troll could report a fake issue saying "Sunny smells" (definitely the most offensive thing they could troll you with) if he knew you were in the room

Reply:
Gotcha. I think we should consider adding more text to better describe what we're expecting the user to do (be more careful next time?) but we can change that later.

Reply:
Though because registration is Waterloo email based, it's trivial to ban someone if they use the system inappropriately. I personally don't think it'll be a major issue - it isn't really worth not being able to book any spaces to troll someone.